### DISCUSSION TITLE
Mask out areas to not be transformed

### DISCUSSION THREAD

**Discussion by gateway** [rkhamilton/vqgan-clip-generator#35] [Ideas]
Is or would it be possible without much work to provide a mask that where the black area is that the style network only works in that area and not in the white?

So with init_image you can provide init_mask and something like this which was an example of my nike shoe..

![image](https://user-images.githubusercontent.com/519563/138614845-bb2e4f9d-0002-462c-8456-f41568d55c46.png)

**Comment by rkhamilton** [comment]
It would be a bit of a research project to figure it out, and I don’t think it would make a clean mask. The stuff happening inside the GAN is a black box. What they are doing inside the black box is: 
1.“make an image”
2.“Does the image look like the prompt?”
3.“Change the parameters to make a new image.”
4. Go to 2.
I can’t readily change what happens when it makes an image, but I can see a path to change the prompt criteria (step 3) to, for example, have a high loss if it changes the area inside a mask. It would be several hours of work, and I’m not sure I would succeed. I also don’t really see what you are trying to do here that’s different than what you’d get from using photoshop. You could always mask out your shoe and layer it onto a generated image. I assume you know that, so I’m guessing you are imagining something different than a composite effect, right?
